newlib/libc/machine/riscv/memcpy.c

```diff
@@ -10,46 +11,101 @@
 */
 
 #if defined(PREFER_SIZE_OVER_SPEED) || defined(__OPTIMIZE_SIZE__)
-//memcpy defined in memcpy-asm.S
+// memcpy defined in memcpy-asm.S
 #else
 
-#include <string.h>
-#include <stdint.h>
 #include "../../string/local.h"
+#include <stdint.h>
+#include <string.h>
 
 #define unlikely(X) __builtin_expect (!!(X), 0)
 
+static inline long
+__load_word (const unsigned char *p)
```

Comment:
this function should be put into the ifdef for slow misaligned access, otherwise you will have an unused function for fast access